librosa/segment.py

```diff
@@ -1138,3 +1176,77 @@ def path_enhance(
         np.clip(R_smooth, 0, None, out=R_smooth)
 
     return R_smooth
+
+
+def __affinity_bandwidth(rec, bw_mode, k):
+
+    # the api allows users to specify a scalar bandwidth directly, besides the string based options.
+    if isinstance(bw_mode, (int, float)):
```

Comment:
This check doesn't seem quite right to me, as we might also want to support a numeric vector input (one bandwidth per point).  We should figure out the right way to identify numeric values including numpy dtypes.

Reply:
Yea... I got lazy here and only supported scalar bandwidth besides the string-based options. So for numeric vector input, we only want to support arrays with shapes the same as `rec.shape` right?